src/array_api_extra/_delegation.py

```diff
@@ -508,3 +508,192 @@ def sinc(x: Array, /, *, xp: ModuleType | None = None) -> Array:
         return xp.sinc(x)
 
     return _funcs.sinc(x, xp=xp)
+
+
+def partition(
+    a: Array,
+    kth: int,
+    /,
+    axis: int | None = -1,
+    *,
+    xp: ModuleType | None = None,
+) -> Array:
+    """
+    Return a partitioned copy of an array.
+
+    Creates a copy of the array and partially sorts it in such a way that the value
+    of the element in k-th position is in the position it would be in a sorted array.
+    In the output array, all elements smaller than the k-th element are located to
+    the left of this element and all equal or greater are located to its right.
+    The ordering of the elements in the two partitions on the either side of
+    the k-th element in the output array is undefined.
+
+    Notes:
+    If `xp` implements `partition` or an equivalent method (e.g. topk for torch),
+    complexity will likely be O(n).
+    If not, this function simply calls `xp.sort` and complexity is O(n log n).
```

Comment:
```suggestion
    If `xp` implements ``partition`` or an equivalent function (e.g. ``topk`` for torch),
    complexity will likely be O(n).
    If not, this function simply calls ``xp.sort`` and complexity is O(n log n).
```

also for `argpartition`